eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,71 @@
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
+$LocalCodeOwnerPath = ".github/CODEOWNERS"
+$ToolCommandName = "retrieve-codeowners"
+$ToolName = "Azure.Sdk.Tools.RetrieveCodeOwners"
+function New-TemporaryDirectory {
+  $parent = [System.IO.Path]::GetTempPath()
+  [string] $name = [System.Guid]::NewGuid()
+  [string] $newPath = Join-Path $parent $name
+  New-Item -ItemType Directory -Path $newPath | Out-Null
+  return $newPath
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+function InstallRetrieveCodeOwnersTool() {
+  # Check if the retrieve-codeowners tool exsits or not.
+  if (Get-Command "$ToolPath/$ToolCommandName" -errorAction SilentlyContinue) {
+    return "$ToolPath/$ToolCommandName"
   }
-}
+  if (!$ToolPath -or !(Test-Path $ToolPath)) {
+    $ToolPath = New-TemporaryDirectory
+  }
+  Write-Warning "Installing the ToolCommandName tool under $ToolPath... "
+  dotnet tool install --tool-path $ToolPath --add-source $DevOpsFeed --version $ToolVersion $ToolName | Out-Null
 
-$results = $ownedFolders[$target]
+  return Join-Path $ToolPath $ToolCommandName
+}
 
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function GetCodeOwners ([string] $Command) {
+  # Run code owner tools to retrieve code owners.
+  if (!(Get-Command $Command -errorAction SilentlyContinue)) {
+    Write-Error "The ToolCommandName tool is not properly installed. Please check your tool path. $ToolPath"
+    return ""
+  }
+  
+  # Params $RootDirectory is already in use in cpp release pipeline. 
+  # Will use $CodeOwnerFileLocation and deprecate $RootDirectory once it is ready to retire $RootDirectory.
+  if ($RootDirectory -and !(Test-Path $CodeOwnerFileLocation)) {
+    $CodeOwnerFileLocation = Join-Path $RootDirectory $LocalCodeOwnerPath
+  }
+  
+  $codeOwnersString = & $Command --target-directory "$TargetDirectory" --code-owner-file-path "$CodeOwnerFileLocation" 
+  # Failed at the command of fetching code owners.
+  if ($LASTEXITCODE -ne 0) {
+    return ""
+  }
+  
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $TargetDirectory"
+    return ""
+  }
   
+  $codeOwners = $codeOwnersJson.Owners -join ","
```

Comment:
If we are only using the joined result inside the if block we should move it in there.